spec/lib/scrabble_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'spec_helper'
+
+describe Scrabble::Scrabble do
+
+  describe "class methods" do
+    it "responds to 'score'" do
+      expect(Scrabble::Scrabble).to respond_to :score
+    end
+
+    context "parameter in score is invalid (nil or "")" do
+      ["", nil].each do |invalid_input|
+        it "returns error message when input is empty or nil" do
+        expect(Scrabble::Scrabble.score(invalid_input)).to eq("ERROR: Invalid input. Either your word has more that 7 letters or you entered something ridiculous. You only have 7 tiles, and Scrabble uses WORDS!")
+        end
+      end
+    end
+
+    context "parameter in score is invalid (number or longer than 7 letters)" do
+      [3423, 2, "elephant"].each do |invalid_input|
+        it "returns error message when input is #{invalid_input}" do
+        expect(Scrabble::Scrabble.score(invalid_input)).to eq("ERROR: Invalid input. Either your word has more that 7 letters or you entered something ridiculous. You only have 7 tiles, and Scrabble uses WORDS!")
+        end
+      end
+
+    end
+
+    [
+      [1, "a"],
+      [1, "e"],
+      [1, "u"],
+      [2, "ae"],
+      [5, "dog"],
+      [3, "DO"],
+      [11, "zip"]
+    ].each do |score, letter|
+      it "returns #{score} for #{letter}." do
+        expect(Scrabble::Scrabble.score(letter)).to eq(score)
+      end
+    end
+
+    it "responds to 'highest_score_from'" do
+      expect(Scrabble::Scrabble).to respond_to :highest_score_from
+    end
+
+    [
+      ["highest", ["a", "an", "dog", "highest"]],
+      ["zipper", ["tan", "zipper", "pants", "hi", "cat"]]
+    ].each do |highest, array_of_words|
+      it "returns #{highest} from #{array_of_words}" do
+        expect(Scrabble::Scrabble.highest_score_from(array_of_words)).to eq(highest)
+      end
+    end
```

Comment:
This is some clever metaprogramming, but I tend to find metaprogrammed tests harder to read. Since this is a unit test for a single method (i.e. `.highest_score_from`), I usually prefer writing one test that describes the functionality of that method. That is, instead of saying something like `it "returns X word from specific array Y" do`, you'd simply say, `it "returns the highest scoring word from an array of word" do`. Then, within that `it` block you can have two `expect` blocks if you really wanted to test more than one, though a single one should usually be enough. If there are unusual sets of word arrays, those should be tested separately as edge cases, with appropriate descriptions.